MapboxDirections/MBRouteLeg.swift

```diff
@@ -43,13 +44,19 @@ open class RouteLeg: NSObject, NSSecureCoding {
                     CongestionLevel(description: $0)!
                 }
             }
+            if let maximumSpeed = jsonAttributes["maxspeed"] as? [JSONDictionary] {
+                maximumSpeedLimits = maximumSpeed.map {
+                    SpeedLimit(json: $0)
+                }
+            }
         }
         
         self.openStreetMapNodeIdentifiers = openStreetMapNodeIdentifiers
         self.segmentDistances = segmentDistances
         self.expectedSegmentTravelTimes = expectedSegmentTravelTimes
         self.segmentSpeeds = segmentSpeeds
         self.segmentCongestionLevels = congestionLevels
+        self.segmentMaximumSpeedLimits = maximumSpeedLimits
```

Comment:
I’m leaning towards calling this property `maximumSegmentSpeedLimits`, which would be more grammatical.

Reply:
This sort of sounds like, the  _maximum number of segments_.